prime-router/docs/universal-pipeline/README.md

```diff
@@ -1,24 +1,262 @@
+
+`@QueueTrigger(name = "message", queueName = elrConvertQueueName)`
+
+`elrConvertQueueName` is set to *elr-fhir-convert*, which can be found in the azure portal amongst other queues:
+
+![azure-queues.png](../assets/azure-queues.png)
+
+When a function has completed successfully, is will use the `QueueAccess` object to send a message to AQS for the next step to run. For example, the Convert step will schedule the Route step for each new Report it created from the original received Report:
+
+```Kotlin
+messagesToSend.forEach { 
+	this.queue.sendMessage(elrRoutingQueueName, it.serialize()) 
+}
+```
+
+In the Convert case above, `messagesToSend` is a list of `RawSubmission` objects which contain the id of the report and some other information that will get sent to the next step in the pipeline. This allows the next step in the pipeline to know which report to work on (`report.id`) and how to get it (`blobInfo.blobUrl`). In the case of [Convert](./convert-translate.md), this code looks like so:
+```Kotlin
+messagesToSend.add(  
+    RawSubmission(  
+        report.id,  
+        blobInfo.blobUrl,  
+        BlobAccess.digestToString(blobInfo.digest),  
+        message.blobSubFolderName,  
+        message.topic  
+    )  
+)
+```
+
+The three queues specific to the Universal Pipeline are:
+- elr-fhir-convert
+- elr-fhir-route
+- elr-fhir-translate
+
+The two queues shared with Legacy Pipeline are:
+- batch
+- send
+
+#### Universal Pipeline Metadata and Report and Item Lineage
+
+> Before continuing with this section, make sure you understand the concept of a [Report and Item](#report-and-item) in ReportStream!
+
+Generally, each step takes in one Report, performs some operation, and outputs a different Report, referred to as a `Child Report`. In some steps, like the [Convert](convert-translate.md) step, the received Report may be split up into multiple child reports! For the Convert step specifically, this is the case when a Report contains multiple items. The Convert step will split the report into multiple, so that each Report contains only one item - this is done to prepare for the [Route](route.md) and [Translate](convert-translate.md) steps.
+
+![item-lineage.png](../assets/item-lineage.png)
+
+Because each step, in general, creates a new Report, it becomes necessary to track the descendants and ancestors of a particular Report to answer important questions such as:
+
+- Where did the submitted Report get sent?
+- Did the submitted Report get delivered?
+
+In order to answer the questions above, each step will record information regarding the actions it took, commonly referred to in ReportStream as `metadata`. The [metadata](../design/metadata.md) is stored in ReportStream's internal [Postgres database](../design/metadata.md) and powers API endpoints like `/history` to help ReportStream clients (senders) answer questions related to the processing of a particular report.
```

Comment:
I get that this is Metadata but might be good to at least mention the action logs and report lineage here

Reply:
I was thinking the metadata document linked here would discuss those concepts in detail. I initially included a blurb about those here but found it to be too much for this document. Thoughts?

Reply:
Lower down I discuss how to get information regarding the status of the report, which is powered by action logs and report/item lineage, but the reader at this level does not need to know that

Reply:
So lower down we say:

The information provided by the history endpoint is captured as the report(s) flow through the pipeline and is referred to as [metadata](file:///Users/arnej/Projects/prime-reportstream/prime-router/docs/design/metadata.md)

and then the user can click metadata link to learn how action logs and report lineage works and powers history API